FCCEUD-27: Describe how to publish a module

theashiot opened the pull request:
Issue ID: FCCEUD-27
Issue Link: https://projects.engineering.redhat.com/browse/FCCEUD-27

* Create module "Publishing a module in Pantheon"

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+[id="publishing-a-module-in-pantheon_{context}"]
+= Publishing a module in {ProductShortName}
+
+Publish a module in {ProductShortName} to the customer portal.
```
s/customer portal/Customer Portal/

theashiot replied:
done

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+.Prerequisites
+
+* You are assigned the publisher role in {ProductShortName}.
+* You are logged-in.
```
s/logged-in/logged in/

theashiot replied:
done

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+
+* You are assigned the publisher role in {ProductShortName}.
+* You are logged-in.
+* The module to be published is uploaded.
```
s/is uploaded/has been uploaded to Pantheon/

theashiot replied:
done

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+
+.Procedure
+
+. Navigate to link:https://pantheon.cee.redhat.com/#/search[].
```
This is an internal URL; we might want to change this to a replaceable such as <pantheonurl> or an attribute for now.

theashiot replied:
changed to a placeholder attribute

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+
+. In the list of module versions, click the three verical dots next to the version of the module you want to publish.
+
+. Click *Edit metadata* to add the following publishing details for the module:
```
I wonder whether adding metadata should be a separate task - this strikes me as something users might do independently of publishing a module. We don't have an issue for that just yet, though, so perhaps we can remove these steps from here but keep the content until we do - would that work for you?

theashiot replied:
Indeed, I checked. The metadata persists across sessions, so it seems like adding metadata is a separate task. I've commented out the step to retain the content.

anarang replied:
Sorry to jump in here but I 'd like to mention that, a user cannot Publish a module until metadata is successfully added. We might want to  consider adding that as a task/step before clicking on Publish button step to let the users know about the expected workflow. Let me know what do you think?

theashiot replied:
Thanks @anarang! 

@adahms, I wonder whether we expect that metadata is already updated when a user is reading these instructions; in that case I can have "metadata is added for the module" as a prerequisite. Any thoughts Anisha?

adahms replied:
Hey @theashiot - thanks for the callout.

I agree - based on Anisha's description, this sounds like a great candidate for a prerequisite. We'll create modules on actually assigning and updating metadata later on, but for now, calling it out as something that needs to happen sounds good to me.

theashiot replied:
Hi @anarang,

I added the following pre-requisite to the module:

"The metadata required for publishing has been applied to the module." 

 Is this ok?

best,
ashwin

anarang replied:
@theashiot sounds good! thanks. :)

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,36 @@
+
+. Click *Publish*.
++
+You can preview the module before publishing by clicking `Preview`.
```
Yes, but we just instructed the user to click 'Publish', so it's too late for this notice. We might create a separate procedure for that, as simple as it might be for now, and include that task as part of a broader user story about preparing content for publication.

theashiot replied:
Agree, and I see Previewing is a part of https://projects.engineering.redhat.com/browse/FCCEUD-28, so removing from here.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,22 @@
+
+.Prerequisites
+
+* You are assigned the publisher role in {ProductShortName}.
```
Same minor notes from above about the 'customer portal', etc., otherwise - looking good!

theashiot replied:
made the changes.

stoobie commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,35 @@
+
+. In the list of module versions, click  *Publish* next to the version of the module you want to publish.
++
+Once published, the module can be accessed at the URL of the product it is published for, followed by its vanity URL fragment.
```
Passive voice, and can be more concise. Change to something like "the module's URL is `https://__PRODUCT_URL__/__VANITY_URL_FRAGMENT__

(Using all caps for variables is based on current style council direction--not yet ratified by department.)

theashiot replied:
Yes, all caps makes sense. Also, I'm not certain if this is true CCS-wide but in EAP docs we use italics for replaceable words. Let me know if the change makes sense.

stoobie replied:
Yes, italics is the current style. The style council proposal is italics and all caps.

stoobie replied:
From commit:

Once published, the module can be accessed at the following URL: =>
After you publish the module, you can access it at the following URL

stoobie replied:
I do see the change in the variable formatting.

theashiot replied:
Once published, the module can be accessed at the following URL: =>
After you publish the module, you can access it at the following URL

Done

stoobie replied:
LGTM

stoobie commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. Click the name of the module you want to unpublish.
++
+If you don't see the module listed, search for the modules using the search box.
```
So you have to know the module's name? There's no browse functionality? Is the search an exact match, or will it provide results if the search term is approximate?

theashiot replied:
Looks like this PR covers searching - https://github.com/redhataccess/pantheon/pull/225

I don't know whether to include details about searching here. How about adding a link to searching modules in additional resources?

theashiot replied:
Added a placeholder to link to Searching Modules under Additional Resources

stoobie replied:
I don't see this in 0a83c34

theashiot replied:
It's there though - https://github.com/redhataccess/pantheon/pull/254/files#diff-df1e50fb25e7b80b7f88efcccc77722fR30

stoobie replied:
LGTM

stoobie commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,22 @@
++
+If you don't see the module listed, search for the modules using the search box.
+
+. Click *Unpublish* next to the module version you want to unpublish.
```
module version => version of the module

Sounds better to me. Suggestion.

theashiot replied:
Agree, updated.

stoobie replied:
I don't see this in 0a83c34.

theashiot replied:
The change is there at https://github.com/redhataccess/pantheon/pull/254/files#diff-63429acfd56fbee86c05f1530069352fR20

stoobie replied:
LGTM

stoobie commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. Click *Unpublish* next to the module version you want to unpublish.
++
+Once a module is unpublished, it is no longer available on the Customer Portal.
```
But unpublish is not delete, right? It's still available in Pantheon? Maybe a short phrase to indicate that would be helpful, especially for the upstream community.

theashiot replied:
Indeed, added a line indicating it is still available in Pantheon.

stoobie replied:
I see this in 0a83c34.

theashiot commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,35 @@
+
+* You are assigned the publisher role in {ProductShortName}.
+* You are logged in.
+* The module to be published has been uploaded to {ProductShortName}.
```
Good point! changed to imported.

stoobie replied:
I don't see this change in 0a83c34.

theashiot replied:
the change is there though - https://github.com/redhataccess/pantheon/pull/254/files#diff-df1e50fb25e7b80b7f88efcccc77722fR11

stoobie replied:
You're right. LGTM.

theashiot commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,35 @@
+[id="publishing-a-module-in-pantheon_{context}"]
+= Publishing a module in {ProductShortName}
+
+Publish a module in {ProductShortName} to the Customer Portal.
```
@adahms can you please provide an insight into this?

adahms replied:
Hey @stoobie @theashiot - good point.

I agree - let's use an attribute for the Customer Portal to represent the publication target. Even downstream, we might end up with several sources, so it could be a good move to future-proof the docs.

theashiot replied:
Added an attribute called PublishingPortal

stoobie replied:
I see this change in 0a83c34f1570f154bf638f6756c1474f2a1d1d70.

theashiot commented:
Hi @stoobie,

I think I have incorporated most of the suggestions. Can you please review the changes?

I've also added a couple of questions inline.

best,
ashwin

theashiot commented:
Hi @stoobie,

Thanks for reviewing! I have made additional change as suggested and linked to the changes that did not appear in the commit - perhaps because I amended the older commit and not create a new commit.

Take a look?

best,
ashwin

stoobie commented:
> Hi @stoobie,
> 
> I think I have incorporated most of the suggestions. Can you please review the changes?
> 
> I've also added a couple of questions inline.
> 
> best,
> ashwin

Hi @theashiot, Everything looks great!

adahms commented:
@redhataccess/pantheon-developers Please review for technical completeness and accuracy.

adahms commented on pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-a-module-in-pantheon.adoc:
```diff
@@ -0,0 +1,44 @@
++
+`https://__PRODUCT_URL__/__VANITY_URL_FRAGMENT__`
++
+where, __PRODUCT_URL__ is the URL for the product, and __VANITY_URL_FRAGMENT__ is the vanity URL fragment for the module.
```
Thanks for the note, @benradey!

@theashiot - I agree that today, it probably makes sense for this to state the automatically generated UUID rather than a vanity URL, especially since we're not covering that exact field in this procedure.

Shall we change it?

theashiot replied:
Thanks @benradey and @adahms, changed the text to:

"https://__PRODUCT_URL__/__UUID__`

where, __PRODUCT_URL__ is the URL for the product, and __UUID__ is the auto-generated unique identifier for the module. "

what do you think?

benradey replied:
Sounds good to me!

theashiot commented:
@redhataccess/pantheon-developers Please merge.

benradey commented:
@theashiot Can do!

I want to point out that according to our process, there should be at least one formal peer approval on the PR before we merge - but it looks like according to the comment chain, both @stoobie and @adahms have looked at this, so I'll assume this is good to go.

I'll get to work on merging this; should only take a few minutes!
